File: Task-4/StringPalindrome.java
public class StringPalindrome {
  public static void main(String[] args){
    System.out.println(isPalindrome1("abcba")); // true
    System.out.println(isPalindrome1("abccba")); // true
    System.out.println(isPalindrome1("abcbbaz")); // false
  }

    //Time Complexity:- O(N)
    public static boolean isPalindrome1(String str){

      boolean check = true;
      int length = str.length();
      int halfLength;

      if( length % 2 == 0 )
        halfLength = length/2;
      else
        halfLength = (length-1)/2;

      //Iterate str till half-length
      for( int i=0; i<halfLength; i++)
        if( str.charAt(i) != str.charAt(length-i-1)){
          check = false;
          break;
        }
      return check;
  }
}

File: Task-4/StringPalindrome.py
def is_palindrome(string):
    return (string==string[::-1])

print("abcba is a palindrome : ", is_palindrome("abcba"))
print("aabccba is a palindrome : ", is_palindrome("abccba"))
print("abcbbaz is a palindrome : ", is_palindrome("abcbbaz"))
